SSLChecker/SSLChecker/main.py

```diff
@@ -0,0 +1,227 @@
+    return _port
+
+
+def verify_scan_type(scan_type:str) -> str:
+    """
+    verify scan type is valid
+    """
+    scan_type = scan_type.lower()
+    if scan_type not in VALID_SCAN_TYPES:
+        raise InvalidRequest(f"Invalid scanner type '{scan_type}'",
+                             ERROR_MSG_INVALID_SCANNER_TYPE)
+    return scan_type
+
+
+def pre_scan_check(req: func.HttpRequest) -> Tuple[str, str, str, int, str]:
+    """
+    return scan_type, view, name, and port, ip as a tuple if the request is
+    valid.
+
+    raises
+    ------
+    InvalidRequest
+    InvalidConfig
+    DNSError
+    """
+    scan_type = req.route_params.get('scan')
+    view = req.route_params.get('view')
+    name = req.route_params.get('name')
+    port = req.route_params.get('port', '443')
+    port = verify_port(port)
+
+    """ Check to ensure ALL parameters were passed in the URI.
+    If you mark the route parameters in function.json as mandatory,
+     the Azure Function worker supplies a 404 if you do not supply all
+     three routes in the URI. I made routes optional, this way we
+     can handle errors gracefully """
+    if scan_type is None or view is None or name is None:
+        raise InvalidRequest("Missing Parameter(s)",
+                             ERROR_MSG_MISSING_PARAMETERS)
+
+    scan_type = verify_scan_type(scan_type)
+    # Check to ensure a valid DNS view was passed
+    view = view.lower()
+    if view not in dnsview:
+        raise InvalidRequest(f"Invalid View '{view}'", ERROR_MSG_INVALID_VIEW)
+
+    # this maybe best handled as part of loading the app instead of checking it
+    # here
+    if dnsview.get(view) == '0.0.0.0':
+        raise InvalidConfig('Missing DNS Server in config',
+                            ERROR_MSG_MISSING_DNS_SERVER)
+
+    # Parse the name parameter to ensure it is a valid DNS name
+    # and does not contain http(s)
+    name = shared_dns.parse_name(name)
+
+    """ Try to resolve the DNS name to an IP to ensure it exists.
+     We use the IP in the scan so that we can record which one we tested
+     which can be useful. """
+    ip = shared_dns.resolve_dns(dnsview.get(view), name)
+
+    return scan_type, view, name, port, ip
+
+def path_params_scanner(req: func.HttpRequest) -> str:
+    """
+    perform actual scan for path based parameters
+    """
+    try:
+        scan_type, view, name, port, ip = pre_scan_check(req)
+        # Run the scan
+        return json.dumps(scanner.scan(name, ip, port, view, scan_type))
+    except (InvalidRequest, InvalidConfig, DNSError, ConnectionError,
+            InvalidFQDN) as err:
+        return json.dumps(results.set_error(err.args[0], err.args[1]))
+    except Exception as err:
+        return json.dumps(results.set_error("Unexpected Error", str(err)))
+
+
+VALID_QUERY_PARAMS = ('host', 'nameserver', 'port', 'scan_type')
+
+ERROR_MSG_QUERY_EXAMPLE = (
+    "Example: https://sslchecker.metlife.com/api/ssl?"

```

Comment:
This is ok I think since its an example, but sslchecker.exampleco.com may be better in the future @gattjoe